gpii/node_modules/processReporter/src/ProcessReporter.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * GPII Process Reporter
+ *
+ * Copyright 2015 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.require("kettle", require);
+fluid.require("./processesBridge.js", require);
+
+fluid.defaults("gpii.processReporter", {
+    gradeNames: "fluid.component"
+});
+
+fluid.defaults("gpii.processReporter.base", {
+    gradeNames: ["kettle.app", "gpii.processReporter"],
+    requestHandlers: {
+        processGet: {
+            route: "/processes",
+            type: "gpii.processReporter.allSolutions.get",
+            method: "get"
+        },
+        processGetSingle: {
+            route: "/processes/:solutionID",
+            type: "gpii.processReporter.solution.get",
+            method: "get"
+        }
+    },
```

Comment:
hey @klown As this is a big part of my improved MM work, I've already merged it in(to my branch) and removed the REST support. Currently, all the needs of the MM has been fulfilled by making direct calls to the PR so I haven't replaced it with anything other than just direct function calls... so for now, I think we can just leave this pull request as is - most of it will be merged in via my pull (once I actually manage to wrap up the MM work) and the remaining parts of the code will be good keep here until we're ready for more advanced usage of the PR